packages/sdk/src/actions/ticket.ts

```diff
@@ -0,0 +1,140 @@
+import { mimc as mimcService } from "../services";
+import { viewAccountHash } from "../views";
+import { sendTransactionsCallback } from "./connection";
+import type { ConnectionType, Currency } from "../interfaces";
+import { getTransaction, randomBN, viewFunction } from "../helpers";
+
+export const createTicket = async (
+  nodeRpcUrl: string,
+  contract: string,
+  accountId: string,
+  currencyId: string
+) => {
+  const { hash } = await mimcService.initMimc();
+
+  const secret = randomBN();
+  const nullifier = randomBN();
+
+  const secretsHash = hash!(secret!, nullifier!);
+
+  const accountHash = await viewAccountHash(nodeRpcUrl, contract, accountId);
+
+  // TODO: check who pass hash and send deposit with hashed value
+  // const currencyHash = await viewAccountHash(
+  //   nodeRpcUrl,
+  //   contract,
+  //   currencyId,
+  // );
+
+  const note =
+    currencyId.toString() +
+    "-" +
+    secret!.toString() +
+    "-" +
+    nullifier!.toString() +
+    "-" +
+    accountHash!.toString();
+
+  return {
+    note,
+    hash: secretsHash,
+  };
+};
+
+export const sendDeposit = async (
+  nodeUrl: string,
+  hash: string,
+  amount: string,
+  depositContract: string,
+  accountId: string,
+  currency: Currency,
+  connection: ConnectionType
+) => {
+  const transactions: any[] = [];
+
+  if (currency.type === "Nep141") {
+    const tokenContract = currency.account_id || "";
+
+    const storage = await getTokenStorage(
+      tokenContract,
+      depositContract,
+      nodeUrl
+    );
+
+    if (!storage || storage.total < "0.10") {
+      transactions.push(
+        getTransaction(
+          accountId,
+          tokenContract,
+          "storage_deposit",
+          {
+            account_id: depositContract,
+            registration_only: true,
+          },
+          "0.50"
```

Comment:
Why 0.5 NEAR? This quantity probably works well, but should be standardized for FT_Storage deposits across entire SDK. Pick a value, create a constant and reuse it